### ISSUE TITLE
Object_detection: Classification loss spikes (10E22) and tfrecord input verification

### ISSUE THREAD

**Issue by karstenBehrendt** [tensorflow/models#3420]
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 14.04, 16.04
- **TensorFlow installed from (source or binary)**: source and binary
- **TensorFlow version (use command below)**: v1.5.0-9-gc959ec71e8 and 1.5
- **Bazel version (if compiling from source)**: 0.10.0
- **CUDA/cuDNN version**: Cuda 8/9, cudnn 7
- **GPU model and memory**: 1080 ti 11GB, X Pascal 12GB
- **Exact command to reproduce**: python object_detection.train.py --train_dir some_dir --pipeline_config_path some_faster_rcnn_config

### Describe the problem

1) **I receive single step spikes in the BoxClassifierLoss randomly which are in the magnitude of up to 10E22**. The dataset only contains one class, the spikes appear using score_converter: SIGMOID, more often with score_converter: SOFTMAX. This happens with both, either num_classes: 1 or 2. Except for those spikes, the classification_loss is very close to zero. Those spikes appear from every 15 samples to every 10,000 samples. Does the loss explode once the logit gets to close to the label? Is there an epsilon added to safeguard logit inputs, is that even needed?
The two following images show spikes for two different runs with different nets.

![classification_loss_spike](https://user-images.githubusercontent.com/4582144/36510843-5054cc92-1719-11e8-994a-3ebcec5b586b.png)
![total_loss_steps](https://user-images.githubusercontent.com/4582144/36510891-77c18f5e-1719-11e8-9846-e7f7aeaa1be3.png)

2) Since I receive these classification loss spikes and no convergence on my own data, **is there a way to check tfrecord inputs for soundness?** The problem could very well be on my side, a typo that I have overlooked while checking or similar. With the spikes and in general, it does not converge and does not show any detections after about 200 samples for a threshold of 0.05. I tried this with several faster rcnn configs, changes are made to the paths, input checkpoints, score_converter (SIGMOID/SOFTMAX), sometimes resizer, and num_classes. The losses seem surprisingly low for now actual detections anywhere. **I would expect one of the configs to detect something, so the error is likely in my tfrecords.**

This appeared with the sample configs:
faster_rcnn_resnet101_coc.config
faster_rcnn_nas_coco.config
faster_rcnn_resnet50_coco_pets.config
Finetune checkpoints are up to date

### Source code / logs
I will attach a sample config, label maps, tfrecord sample image output, images of loss, sample loss log
A quick check that my tfrecord values are within bounds, and part of the label maps
[tfrecord_value_checks.txt](https://github.com/tensorflow/models/files/1746068/tfrecord_value_checks.txt)

Label maps [vehicles_complete.txt](https://github.com/tensorflow/models/files/1746077/vehicles_complete.txt)

Shortened tfrecords can be found at 
https://www.dropbox.com/s/by4ws4ojb69dhvw/complete_test.tfrecords?dl=0
https://www.dropbox.com/s/juxllgbs21lkll8/complete_train.tfrecords?dl=0
They are limited to 500 samples each, approx 200MB each.

I hope all of the data provided is enough to start a test run in under 10 minutes.

Attachments:
Sample image in tfrecord with (black) bounding boxes
![sample_image](https://user-images.githubusercontent.com/4582144/36510916-9356090c-1719-11e8-8baf-f417f4bdb338.png)

Other losses:
![box_localization_loss](https://user-images.githubusercontent.com/4582144/36510952-b4efba72-1719-11e8-8057-c1d089a446de.png)
![rpn_localization_loss](https://user-images.githubusercontent.com/4582144/36510953-b812aaca-1719-11e8-8628-c918b3325137.png)
![rpn_objectness_loss](https://user-images.githubusercontent.com/4582144/36510955-badb9816-1719-11e8-883d-a986042d82fe.png)
![total_loss](https://user-images.githubusercontent.com/4582144/36510957-bf3ce716-1719-11e8-949d-0d0773d8938b.png)

What should I check next? What more information would be helpful? 
Thanks very much for creating this framework

**Comment by asimshankar** [comment]
Thanks for the note, however we probably don't have the bandwidth to investigate and debug your input data.

If you could distill this into a more specific bug / feature request we'd be happy to take a look.
Alternatively, if you may want to seek support on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since there is also a larger community that reads questions there. Though again, you'd probably want to do some more investigation to seek more specific help.

**Comment by Slaybesh** [comment]
I'm having the same issue, have you solved the issue?

**Comment by karstenBehrendt** [comment]
My issue was that one label name in the pbtxt did not have the same name as the label-text in the tfrecord. Since the name in the label map did not exist, it was assigned a "-2" as label id which messed 
 up the training.

I assumed that the label id in the tfrecord would be used for training and that the text would be ignored at that stage. Digging through the code I saw that the label text is preferred over the id.

**Comment by Slaybesh** [comment]
Thanks very much! What I had proposed earlier was wrong so I removed it.
That definitely solved the issue.

**Comment by VenkatGudala** [comment]
@karstenBehrendt 
Hi, I have the same issue of loss explosion after 300 steps of training and come across this thread in github. would you help me explaining how you resolved your issue. I checked my csv files of train and test and cross checked the labels, and ensured that no xmin, ymin < 0 and height, width > ymax, xmax. can you point me to the direction to resolve my issue?

**Comment by karstenBehrendt** [comment]
@VenkatGudala My response from March 7, 2018 is all I got. There is no way for me to know what your issue is
